src/dtx/dtx_common.c

```diff
@@ -297,34 +297,51 @@ dtx_begin(struct dtx_id *dti, daos_unit_oid_t *oid, daos_handle_t coh,
 }
 
 int
-dtx_end(struct dtx_handle *dth, struct ds_cont_hdl *cont_hdl,
-	struct ds_cont *cont, int result)
+dtx_end(struct dtx_handle *dth, dtx_wait_cb_t wait_cb,
+	struct ds_cont_hdl *cont_hdl, struct ds_cont *cont, int result)
```

Comment:
why do we want to call wait_cb() within here?  
We could explicitly call dispatch_wait() before this function right? It has nothing to do with DTX

Reply:
I think wait all parities to finish its execution is exactly dtx logic. so it seems to me wait should be done in dtx_end(). Please check my next patch.